src/index.js

```diff
@@ -468,14 +482,18 @@ class Masq {
     if (!this.dataEncryptionKey) throw Error('Data encryption key is not set')
   }
 
+  _checkNonce () {
+    if (!this.nonce) throw Error('Nonce is not set')
+  }
+
   /**
    * Set a watcher
    * @param {string} key - Key
    * @returns {Object}
    */
-  watch (key, cb) {
+  async watch (key, cb) {
     const db = this._getDB()
-    return db.watch(key, cb)
+    await common.utils.watch(db, this.nonce, key, cb)
```

Comment:
I don't think watch is an async function

Reply:
yes but we are calling hashKey inside watch method, and this call is async.

Reply:
ok but this will be proceesed inside the callback IIUC ?

Reply:
I don't understand what callback IIUC is ?
But the hashKey function is called before the actual 'not async' method watch on hyperdb

Reply:
lol IIUC stand for "If I Understand Correctly" :rofl:

Reply:
that's true, we need to wait for `hashKey`, so it seems we need to await this watch method